src/design-pages/passkey_kerberos.rst

```diff
@@ -0,0 +1,309 @@
+  in AS-REP.
+
+* krb5_child stores the ticket and return the authentication success to the IPA
+  provider. This information is propagated by the PAM responder, PAM and
+  finally the user.
+
+Data structures
+***************
+This section defines the data needed to obtain and verify the assertion.
+
+Assertion request data
+++++++++++++++++++++++
+To obtain the assertion the following information is required:
+
+* Domain (String)
+
+* List of credential IDs in b64 (String)
+
+* User-verification (Int)
+
+* Cryptographic challenge (String)
+
+.. _assertion-data:
+
+Assertion data
+++++++++++++++
+To verify the assertion the following informations is required:
+
+* Username (String)
+
+* Domain (String)
+
+* Used credential ID in b64 (String)
+
+* Public key in b64 (String)
+
+* COSE type (String)
+
+* User-verification (Int)
+
+* Cryptographic challenge (String)
+
+* Authenticator data in b64 (String)
+
+* Assertion signature in b64 (String)
+
+.. _kerberos-kdc-ipa-otpd:
+
+Kerberos KDC - ipa-otpd
++++++++++++++++++++++++
+The Kerberos KDC is expected to provide the following json-formatted string
+when generating the first Access-Request:
+
+..  code-block:: json
+
+    "passkey": {
+        "phase": 0
+    }
+
+ipa-otpd is expected to return a json-formatted string with the assertion
+request data:
+
+..  code-block:: json
+
+    "passkey": {
+        "phase": 1,
+        "state": "$ipa_otpd state",
+        "data": {
+            "domain": "$domain",
+            "credential_id": "$credential_id_list",
+            "user_verification": "$user_verification",
+            "cryptographic_challenge": "$cryptographic_challenge"
+        }
+    }
+
+The second Access-Request generated by the Kerberos KDC should contain the
+following json-formatted string in the password field:
+
+..  code-block:: json
+
+    "passkey": {
+        "phase": 2,
+        "state": "$ipa_otpd state",
+        "data": {
+            "credential_id": "$credential_id",
+            "cryptographic_challenge": "$cryptographic_challenge",
+            "authenticator_data" :"$authenticator_data",
+            "assertion_signature": "$assertion_signature",
+            "user_id": "$user_id"
+        }
+    }
+
+ipa-otpd - passkey_helper
++++++++++++++++++++++++++
+The passkey_helper process located in the FreeIPA server needs to regenerate
+the assertion data to be validated by the passkey. ipa-otpd executes the
+passkey_helper and thus, it is in charge of providing the assertion data as
+:ref:`previously specified<assertion-data>`.
+
+Configuration options
+*********************
+
+sssd
+++++
+
+The following domain option can be used to tune the authentication policy:
+
+.. _local-auth-policy:
+
+* local_auth_policy: local authentication methods policy. Some backends (i.e.
+  LDAP, proxy provider) only support a password base authentication, in such
+  cases authentication is only performed with the methods supported by the
+  backend.
```

Comment:
Hi,

I think in this paragraph some additional context would be beneficial. E.g.
```
Some backends (i.e. LDAP, proxy provider) only support a password base authentication while other can handle PKINIT based Smartcard authentication (AD, IPA), two-factor authentication (IPA) or other methods against a central instance. By default in such cases authentication is only performed with the methods supported by the backend. To allow more convenient or secure authentication methods which are supported by SSSD but not by the backend in cases where a central authentication is not strictly required the `local_auth_policy` option is added.
```

bye,
Sumit

Reply:
I included this paragraph. Thank you!